### ISSUE TITLE
Referencing outputs from one stack in another stack's configuration

### ISSUE THREAD

**Issue by ethernet-zero** [pulumi/pulumi#9112]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

My team and I manage a few projects in the cloud, and we use Pulumi to provision the cloud infrastructure. The way we organize our IaC code is by breaking it down in several repos, each one taking care of one part of the infra. For example, we have one project for the networking stuff, VPCs and such. Another one for Kubernetes clusters, another one for database stuff, something like that. The way we keep all of those projects under the same umbrella is by setting the same project name on all of them, so each one has its own stacks for the different environments but all of them are under the same project. Also, all of those stacks are in the same backend.

Lately, I was wondering if there could be a way of referencing an output exported in one of the stacks directly from the configuration of another stack in the same project. Basically, something like creating a `StackReference` but being able to do it directly from the stack configuration. Here's a use case: let's say we have a stack named `myproject/network/dev` where we create a VPC in Google Cloud and we export the self link to that VPC:

```python
# We use Python, by the way, but it's not relevant.
pulumi.export('vpc', my_vpc.self_link)
```

Then we have a GKE stack named `myproject/kubernetes/dev`, and we need to tell it in what VPC should Google Cloud place the Kubernetes cluster, so we have a config key for that:

```yaml
myproject:cluster_vpc: myproject-myvpc-c4f3b4b3
```

It would be really cool and convenient if we could directly reference that exported value from the other stack, for example by defining a YAML tag for that purpose:

```yaml
# This is an example: !output <name of the stack> <name of the exported output>
myproject:cluster_vpc: !output myproject/network/dev vpc
```

And that way, when reading that config key:

```python
cfg = pulumi.Config()
vpc = cfg.get('cluster_vpc') # Gets the self_link output of the VPC
```

Having that tag in the stack config would add an automatic stack reference to the state.

We have implemented something like that in our projects, but it'd be so cool if Pulumi had support for that built in. I'd even vouch for submitting a PR myself if I knew where exactly to place the code to add the custom tag resolver and serializer.

Do you think a feature like that could make sense and not bring up a ton of issues?

### Affected area/feature

Stack configuration I guess? I don't really know. :stuck_out_tongue:

**Comment by blamarvt** [comment]
I will say this is a huge blocker for me recommending Pulumi over Terraform Cloud as with TF cloud you can reference remote state from different workspaces very easily.

**Comment by Yavis** [comment]
This feature actually got implemented I guess?
See: https://www.pulumi.com/docs/iac/concepts/stacks/

**Comment by Frassle** [comment]
> This feature actually got implemented I guess?

No. There's currently no way to set config to use another stacks output yet.
